server/sonar-server/src/main/java/org/sonar/server/user/ws/GroupsAction.java

```diff
@@ -20,115 +20,121 @@
-    int page = request.mandatoryParamAsInt(Param.PAGE);
-    String queryString = request.param(Param.TEXT_QUERY);
-    String selected = request.mandatoryParam(Param.SELECTED);
-
-    GroupMembershipQuery query = GroupMembershipQuery.builder()
-      .login(login)
-      .groupSearch(queryString)
-      .membership(getMembership(selected))
-      .pageIndex(page)
-      .pageSize(pageSize)
-      .build();
+  private GroupsWsResponse doHandle(GroupsRequest request) {
 
     try (DbSession dbSession = dbClient.openSession(false)) {
-      UserDto user = dbClient.userDao().selectByLogin(dbSession, login);
-      if (user == null) {
-        throw new NotFoundException(String.format("User with login '%s' has not been found", login));
-      }
+      OrganizationDto organization = findOrganizationByKey(dbSession, request.getOrganization());
+      String login = request.getLogin();
+      GroupMembershipQuery query = GroupMembershipQuery.builder()
+        .organizationUuid(organization.getUuid())
+        .groupSearch(request.getQuery())
+        .membership(getMembership(request.getSelected()))
+        .pageIndex(request.getPage())
+        .pageSize(request.getPageSize())
+        .build();
+      UserDto user = checkFound(dbClient.userDao().selectActiveUserByLogin(dbSession, login), "Unknown user: %s", login);
       int total = dbClient.groupMembershipDao().countGroups(dbSession, query, user.getId());
-      Paging paging = forPageIndex(page).withPageSize(pageSize).andTotal(total);
-      List<GroupMembershipDto> groups = dbClient.groupMembershipDao().selectGroups(dbSession, query, user.getId(), paging.offset(), pageSize);
-
-      JsonWriter json = response.newJsonWriter().beginObject();
-      writeGroups(json, groups);
-      writePaging(json, paging);
-      json.endObject().close();
+      Paging paging = forPageIndex(query.pageIndex()).withPageSize(query.pageSize()).andTotal(total);
+      List<GroupMembershipDto> groups = dbClient.groupMembershipDao().selectGroups(dbSession, query, user.getId(), paging.offset(), query.pageSize());
+      return buildResponse(groups, paging);
     }
   }
 
-  private static void writeGroups(JsonWriter json, List<GroupMembershipDto> groups) {
-    json.name("groups").beginArray();
-    for (GroupMembershipDto group : groups) {
-      json.beginObject()
-        .prop(FIELD_ID, group.getId().toString())
-        .prop(FIELD_NAME, group.getName())
-        .prop(FIELD_DESCRIPTION, group.getDescription())
-        .prop(FIELD_SELECTED, group.getUserId() != null)
-        .endObject();
-    }
-    json.endArray();
+  private OrganizationDto findOrganizationByKey(DbSession dbSession, @Nullable String key) {
+    String effectiveKey = key == null ? defaultOrganizationProvider.get().getKey() : key;
+    Optional<OrganizationDto> org = dbClient.organizationDao().selectByKey(dbSession, effectiveKey);
+    checkFoundWithOptional(org, "No organization with key '%s'", key);
+    return org.get();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Call "org.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)